core/shared/src/main/scala/zio/URIO.scala

```diff
@@ -198,6 +198,18 @@ object URIO {
   def collectAllWithParN[R, A, B](n: Int)(as: Iterable[URIO[R, A]])(f: PartialFunction[A, B]): URIO[R, List[B]] =
     ZIO.collectAllWithParN(n)(as)(f)
 
+  /**
+   * @see See [[zio.ZIO.collectSome]]
+   */
+  def collectSome[R, E, A, B](in: Iterable[A])(f: A => ZIO[R, Option[E], B]): ZIO[R, E, List[B]] =
```

Comment:
For the aliases on the companion object the contract is that it should give us the more specific type. So for example since this is defined on `URIO` we should drop the `E` type parameter, `f` should be `A => URIO[R, Option[Nothing], B]` and the return type should be `URIO[R, List[B]]`.